File: cpp_version/Wilk.cpp
#include "Wilk.h"

Wilk::Wilk(int x, int y, Swiat& swiat)
    : Zwierze(9, 5, x, y, swiat) {
} 

char Wilk::rysowanie() const {
    return 'W'; 
}

bool Wilk::czyTenSamGatunek(Organizm& inny) const {
    return dynamic_cast<const Wilk*>(&inny) != nullptr;
}


std::string Wilk::getNazwa() const {
    return "Wilk";
}

std::shared_ptr<Organizm> Wilk::stworzPotomka(int nowyX, int nowyY) {
    return std::make_shared<Wilk>(nowyX, nowyY, swiat);
}

File: java_version/Wilk.java
public class Wilk extends Zwierze {
    public Wilk(int x, int y, Swiat swiat) {
        super(9, 5, x, y, swiat);
    }

    @Override
    public char rysowanie() {
        return 'W';
    }

    @Override
    public boolean czyTenSamGatunek(Organizm inny) {
        return inny instanceof Wilk;
    }

    @Override
    public String getNazwa() {
        return "Wilk";
    }

    @Override
    public Organizm stworzPotomka(int nowyX, int nowyY) {
        return new Wilk(nowyX, nowyY, swiat);
    }
}
